internal/util/argParser.go

```diff
@@ -122,6 +99,50 @@ func CheckForMissingValues(arguments []string) error {
 
 	return nil
 }
+func FilterServiceKind(input []Input) (string, error) {
+	serviceKindInput := filterInput(input, func(input Input) bool {
+		return input.ParameterHierarchy[0] == K8S_SERVICE_KIND
+	})
+	if len(serviceKindInput) != 1 {
+		return "", fmt.Errorf("ServiceKind is missing")
+	} else {
+		return serviceKindInput[0].Value, nil
+	}
+}
+
+func filterInput(slice []Input, predicate func(Input) bool) []Input {
+	var result []Input
+	for _, item := range slice {
+		if predicate(item) {
+			result = append(result, item)
+		}
+	}
+	return result
+}
```

Comment:
Did you reimplement [slices.IndexFunc](https://pkg.go.dev/slices@go1.21.0#IndexFunc)? :D

Reply:
Not exactly, but yes ':)